### PR TITLE
comp(typeahead): return selected object

### PR DESCRIPTION
**Pull request by miqmago**
Return selected object (in case of object sources) instead of selected string

### CODE REVIEW

File: components/typeahead/typeahead.ts [hunk 1]
```diff
@@ -141,9 +146,15 @@ export class TypeaheadContainer {
     return queryToEscape.replace(/([.?*+^$[\]\\(){}|-])/g, '\\$1');
   }
 
-  private hightlight(item:string, query:string) {
+  private hightlight(item:any, query:string) {
+    var itemStr:string;
```
Comment (valorkin) [inline components/typeahead/typeahead.ts]: @miqmago I am really against `if else` statements
can we simplify to

``` js
var itemStr = (typeof item === 'object' && this._field ? item[this._field] : item).toString()
```

File: components/typeahead/typeahead.ts [hunk 2]
```diff
@@ -393,12 +404,18 @@ export class Typeahead implements OnInit {
   }
 
   public changeModel(value:any) {
-    this.cd.viewToModelUpdate(value);
-    setProperty(this.renderer, this.element, 'value', value);
+    var valueStr:string;
```
Comment (valorkin) [inline components/typeahead/typeahead.ts]: same here

**Comment by miqmago** [comment]
Done.

I've also completely rewritten some parts, in order to achive "google style" matching, implementing "latinize" to match `súper` with `super` (and vice versa) and break words with spaces (`"exact phrase" here match` would match with `match exact phrase here` but not with `phrase here exact match` - kind of "google style"). I've tested and the speed is fine. Also solves the problem with upper case hightlight.

I can make the commits here or wait to open a new PR if you are interested. (I prefer step by step).

**Comment by valorkin** [comment]
LGTM!

**Comment by AlonsoK28** [comment]
> Done.
> 
> I've also completely rewritten some parts, in order to achive "google style" matching, implementing "latinize" to match `súper` with `super` (and vice versa) and break words with spaces (`"exact phrase" here match` would match with `match exact phrase here` but not with `phrase here exact match` - kind of "google style"). I've tested and the speed is fine. Also solves the problem with upper case hightlight.
> 
> I can make the commits here or wait to open a new PR if you are interested. (I prefer step by step).

Excellent

How can I use TypeHeadUtils to search inside an literal object using your Algorithms (for example Latinize and others) and not to be rewrite a completly new code from scratch 💯 💯💯💯💯💯


For example like this:

```
const search ='súper';
const typeHeadConfig = {latinize: true};
const res = myObject.typeHeadUtils.search(search, typeHeadConfig);
console.log(res) //prints súper and super inside an array with other results
```

**Pull request merged**